test/e2e/scale.go

```diff
@@ -200,68 +228,56 @@ func ScaleSpec(ctx context.Context, inputGetter func() ScaleSpecInput) {
+				"true if useCrossNamespaceClusterClass is true")
 		}
 
-		machineDeploymentCount := ptr.To[int64](1)
-		if input.MachineDeploymentCount != nil {
-			machineDeploymentCount = input.MachineDeploymentCount
-		}
-		// If variable is defined that will take precedence.
-		if input.E2EConfig.HasVariable(scaleMachineDeploymentCount) {
-			machineDeploymentCountStr := input.E2EConfig.GetVariable(scaleMachineDeploymentCount)
-			machineDeploymentCountInt, err := strconv.Atoi(machineDeploymentCountStr)
-			Expect(err).ToNot(HaveOccurred())
-			machineDeploymentCount = ptr.To[int64](int64(machineDeploymentCountInt))
-		}
-
-		clusterCount := int64(10)
-		if input.ClusterCount != nil {
-			clusterCount = *input.ClusterCount
-		}
-		// If variable is defined that will take precedence.
-		if input.E2EConfig.HasVariable(scaleClusterCount) {
-			clusterCountStr := input.E2EConfig.GetVariable(scaleClusterCount)
-			var err error
-			clusterCount, err = strconv.ParseInt(clusterCountStr, 10, 64)
-			Expect(err).NotTo(HaveOccurred(), "%q value should be integer", scaleClusterCount)
-		}
-
-		concurrency := int64(5)
-		if input.Concurrency != nil {
-			concurrency = *input.Concurrency
-		}
-		// If variable is defined that will take precedence.
-		if input.E2EConfig.HasVariable(scaleConcurrency) {
-			concurrencyStr := input.E2EConfig.GetVariable(scaleConcurrency)
-			var err error
-			concurrency, err = strconv.ParseInt(concurrencyStr, 10, 64)
-			Expect(err).NotTo(HaveOccurred(), "%q value should be integer", scaleConcurrency)
+		if input.ExtensionServiceNamespace != "" && input.ExtensionServiceName != "" {
+			// NOTE: test extension is already deployed in the management cluster. If for any reason in future we want
+			// to make this test more self-contained this test should be modified in order to create an additional
+			// management cluster; also the E2E test configuration should be modified introducing something like
+			// optional:true allowing to define which providers should not be installed by default in
+			// a management cluster.
+			By("Deploy Test Extension ExtensionConfig")
+
+			// In this test we are defaulting all handlers to non-blocking because we don't expect the handlers to block the
+			// cluster lifecycle by default. Setting defaultAllHandlersToBlocking to false enforces that the test-extension
+			// automatically creates the ConfigMap with non-blocking preloaded responses.
+			defaultAllHandlersToBlocking := false
+			// select on the current namespace if the Clusters are all deployed in the current namespace
+			// This is necessary so in CI this test doesn't influence other tests by enabling lifecycle hooks
+			// in other test namespaces.
+			selectNamespace := !*input.DeployClusterInSeparateNamespaces
+			Expect(input.BootstrapClusterProxy.GetClient().Create(ctx,
+				extensionConfig(specName, namespace.Name, input.ExtensionServiceNamespace, input.ExtensionServiceName, selectNamespace, defaultAllHandlersToBlocking))).
```

Comment:
specName as name is hardcoded to scale from this file here. Wondering if we should allow to modify this?

Reply:
Thx, added an input parameter to both tests for this